intake_solr/source.py

```diff
@@ -108,32 +150,42 @@ class SOLRTableSource(SOLRSequenceSource):
     zoocollection: bool or str
         If using Zookeeper to orchestrate SOLR, this is the name of the
         collection to connect to.
+    partition_len: int or None
+        The desired partition size. [default: 1024]
     """
 
     name = 'solrtab'
     container = 'dataframe'
+    partition_access = True
 
     def _get_schema(self, retry=2):
         """Get schema from first 10 hits or cached dataframe"""
-        if not hasattr(self, '_dataframe'):
-            self._get_partition(0)
-        dtype = {k: str(v)
-                 for k, v in self._dataframe.dtypes.to_dict().items()}
-        return base.Schema(datashape=None,
-                           dtype=dtype,
-                           shape=self._dataframe.shape,
-                           npartitions=1,
-                           extra_metadata={})
-
-    def _get_partition(self, _):
-        """Downloads all data
+        schema = super()._get_schema()
+
+        df = self._get_partition(0)
```

Comment:
This loads the first partition into a dataframe and uses it to discover the returned schema.
Note that the schema might be different from the overall Solr core schema because the user can select a subset of fields using the `fl` `qarg`.